pgmpy/factors/CPD.py

```diff
@@ -116,15 +119,33 @@ def __init__(self, variable, variable_card, values,
 
         super(TabularCPD, self).__init__(variables, cardinality, values.flatten('C'))
 
-    def get_cpd(self):
+    def get_cpd(self, pretty=True):
         """
         Returns the cpd
         """
         if self.variable in self.variables:
+            if pretty:
+                table_list = []
+                for i in range(len(self.cardinality))[1:]:
+                    var_rep = np.prod(self.evidence_card[i:])
+                    row = [self.scope()[-i]]
+                    row.extend(self.variables[row[0]] * var_rep)
+                    if i > 1:
+                        row[1:] = [x for item in row[1:] for x in
+                                   repeat(item, int(np.prod(self.evidence_card[0:i - 1])))]
```

Comment:
Here I need to convert an arrays like ```["diff","diff_0","diff_1"]``` into ```["diff","diff_0","diff_0","diff_1","diff_1"]```.

This was what I found after some pondering. 
Any more pythonic way to achieve something like this in general.?